src/peep/peep.c

```diff
@@ -6160,14 +6171,15 @@ void peep_set_map_tooltip(rct_peep *peep)
 
 
 void sub_693BAB(rct_peep* peep) {
-	uint8 bl = peep->var_6F;
-	if (bl != peep->action_sprite_type) {
+	// Add spriteType as function parameter and make peep->next_action_sprite_type obsolete?
+	uint8 spriteType = peep->next_action_sprite_type;
+	if (spriteType != peep->action_sprite_type) {
 		invalidate_sprite_2((rct_sprite*)peep);
-		peep->action_sprite_type = bl;
+		peep->action_sprite_type = spriteType;
 		uint8* edx = RCT2_ADDRESS(0x98270C, uint8*)[peep->sprite_type * 2];
```

Comment:
`0x00982708` is a large array that contains the following information `base_image_id, sprite_width, sprite_height_negative, sprite_height_positive, + 1 uint8 not used` perhaps we should define that array.